bgp/modules/terms.py

```diff
@@ -353,6 +353,32 @@ def __init__(self):
         super().__init__(['copyright', '©'], extractor=self.extractor, match_limit=1)
 
 
+class TocPageDetectorModule(KeywordPageDetectorModule):
+
+    def __init__(self):
+        super().__init__(["table of content"], match_limit=1)
+
+    def detectTocHeading(self, page):
+        for line in page.iter('LINE'):
+            heading = ""
+            for word in line.iter('WORD'):
+                heading = heading + " " + word.text
+            if heading.lower().strip() in self.keywords:
+                return True
+        
+        return False
```

Comment:
This is an interesting example of where things may break:
https://archive.org/details/manualzilla-id-5654336/page/n1/mode/2up?q=table+of+contents

Here are some possible simplfiications

```suggestion
        for i, line in enumerate(page.iter('LINE')):
            if i < 5:  # if we're in the first few lines
            words = " ".join(line.iter('WORD')).lower().strip()
            if any(kws == words for kws in self.keywords):
                return True
        return False
```

Reply:
This edge-case totally skipped through my mind. The change you have made seems perfect to me. However as you later suggested that for the first version we should only check "Table of contents", if we do that then for this book we will directly detect [this](https://archive.org/details/manualzilla-id-5654336/page/n3/mode/2up?q=table+of+contents&view=theater) instead of [this](https://archive.org/details/manualzilla-id-5654336/page/n1/mode/2up?q=table+of+contents&view=theater) because the page where the sections are mentioned(second link) uses "Contents" and the page where the table of contents for a particular section is mentioned (first link) uses "Table of Content". Other than that it seems perfect to me.